site/app/views/submission/HomeworkView.php

```diff
@@ -558,7 +559,13 @@ public function getRandomIndices(int $array_length, string $student_id, string $
             'is_grader_view' => false,
             'recent_version_url' => $recent_version_url,
             'git_auth_token_url' => $this->core->buildUrl(['authentication_tokens']),
-            'git_auth_token_required' => false
+            'git_auth_token_required' => false,
+            'vcs_type_none' => GradeableUtils::VCS_TYPE_NONE,
+            'vcs_type_submitty_hosted' => GradeableUtils::VCS_TYPE_SUBMITTY_HOSTED,
+            'vcs_type_submitty_hosted_url' => GradeableUtils::VCS_TYPE_SUBMITTY_HOSTED_URL,
+            'vcs_type_public_github' => GradeableUtils::VCS_TYPE_PUBLIC_GITHUB,
+            'vcs_type_private_github' => GradeableUtils::VCS_TYPE_PRIVATE_GITHUB,
+            'vcs_type_self_hosted' => GradeableUtils::VCS_TYPE_SELF_HOSTED
```

Comment:
There should be a better way to pass these, though I'm not sure exactly how to do it. I do think these keys should be uppercase too since they're constants.

Reply:
Ok, I changed the constants to the const(namespace/class::const_name) method